tests/test_decision_service.py

```diff
@@ -1421,3 +1390,238 @@ def test_get_variation_for_feature__returns_none_for_user_in_group_experiment_no
+        mock_generate_bucket_value.assert_called_with('test_user111135')
+
+    def test_get_variation_for_feature__returns_variation_for_feature_in_experiment_bucket_range_5000_7500(
+        self,
+    ):
+        """ Test that if a user is in the non-mutex group and the user bucket value should be equal to 5000
+        or greater than 7500."""
+
+        feature = self.project_config.get_feature_from_key("test_feature_in_multiple_experiments")
+        expected_experiment = self.project_config.get_experiment_from_key("test_experiment5")
+        expected_variation = self.project_config.get_variation_from_id(
+            "test_experiment5", "222241"
+        )
+        user_attr = {"experiment_attr": "group_experiment"}
+
+        with mock.patch(
+            'optimizely.bucketer.Bucketer._generate_bucket_value', return_value=6500) as mock_generate_bucket_value,\
+                mock.patch.object(self.project_config, 'logger') as mock_config_logging:
+
+            variation_received, _ = self.decision_service.get_variation_for_feature(
+                self.project_config, feature, "test_user", user_attr
+            )
+            self.assertEqual(
+                decision_service.Decision(
+                    expected_experiment,
+                    expected_variation,
+                    enums.DecisionSources.FEATURE_TEST,
+                ),
+                variation_received,
+            )
+        mock_config_logging.debug.assert_called_with('Assigned bucket 6500 to user with bucketing ID "test_user".')
+        mock_generate_bucket_value.assert_called_with('test_user111136')
+
+    def test_get_variation_for_feature__returns_variation_for_rollout_in_experiment_bucket_greater_than_7500(
+            self,
+    ):
+        """ Test that if a user is in the non-mutex group and the user bucket value should be greater than  7500."""
+
+        feature = self.project_config.get_feature_from_key("test_feature_in_multiple_experiments")
+        user_attr = {"experiment_attr": "group_experiment"}
+        with mock.patch(
+            'optimizely.bucketer.Bucketer._generate_bucket_value', return_value=8000) as mock_generate_bucket_value, \
+                mock.patch.object(self.project_config, 'logger') as mock_config_logging:
+            variation_received, _ = self.decision_service.get_variation_for_feature(
+                self.project_config, feature, "test_user", user_attr
+            )
+            self.assertEqual(
+                decision_service.Decision(
+                    None,
+                    None,
+                    enums.DecisionSources.ROLLOUT,
+                ),
+                variation_received,
+            )
+
+        mock_generate_bucket_value.assert_called_with('test_user211147')
+        mock_config_logging.debug.assert_called_with('Assigned bucket 8000 to user with bucketing ID "test_user".')
```

Comment:
These tests look great. 
One missing test is that in multiple experiments, one of the test can be set to missing target by audience (instead of traffic allocation). We do not need to repeat multiple traffic allocation tests. One of them can be changed to use audience-mismatch.